projects/verkle-trie-migration-exploration.md

```diff
@@ -0,0 +1,97 @@
+
+## Project description
+
+I’m a computer engineer that likes the sweet spot between engineering and research. My intuition before starting this project was that there were probably a lot of things to help with:
+
+- The switch of cryptographic primitives can incur new computational costs that can affect validators’ hardware. I’ll dive into building reference benchmarks to understand this cost and how much it can be improved compared to other state-of-the-art implementations.
+- Doing a data structure migration isn’t only about a spec on how the new data structure works but also is an engineering problem on how to do the switch without impact since there’s an existing state of being migrated. I’ll try to understand more about how this *switch* will happen and have an idea of the risks.
+- The people working on these today have done great talks and blog posts that describe what this is all about. I still think people should understand more about how this works and why it is important. I’m open to help in writing documentation or technical blog posts to raise awareness of why this work is important. It can also bring more attention so more people can help.
+
+I’ll have an open mind in this project, so I can ask or sense my mentor's current pain points and help them if possible. I’ll try to balance doing exploratory work and mainstream work that can be useful today and in the future.
+
+## Specification
+
+Two main forces will guide my work:
+
+- My intuition as an engineer and amateur researcher is to build a mental model of the problems and solutions.
+- Talking with my mentors to understand their current challenges and verify my understanding.
+
+My solution will consist of the following:
+
+- Detailed public documents (e.g., HackMD docs) that will do an in-depth analysis of details about the problem to be solved. This helps document and onboard more people into current affairs and challenges.
+- Take a data-driven approach to validate hypotheses and bottlenecks. I’ll create reproducible benchmarks available in open-source repos that anyone can verify and reproduce. The idea is that all the work can keep evolving and isn’t lost in some code snippet.
+- Open PRs into relevant repos to help with needed work (e.g: [go-ethereum](https://github.com/ethereum/go-ethereum), [go-verkle](https://github.com/gballet/go-verkle), [go-ipa](https://github.com/crate-crypto/go-ipa))
+- Open discussion with mentors in #verkle-trie-migration so we can have a history of discussions and help raise awareness of some of the current work with Verkle Tries.
+
+## Roadmap
+
+Since there’s a lot of work happening quite fast in this area, it’s a bit hard to have a meaningful and stable roadmap; but I’d imagine the following steps:
+
+- Gain an understanding of Verkle Tries specification and new cryptographic primitives.
```

Comment:
You can also make these check boxes for continues tracking :)

Reply:
Nice, I've updated these to todo-bullets. Thanks for the idea!